### PR TITLE
uml-init: remove host systemd dependency

### PR DESCRIPTION
**Pull request by mtth-bfft**
The current UML workflow depends on the host system init being systemd:
- boot UML with host filesystem and the default init=/sbin/init
- otherwise, systemd parses the SYSTEMD_UNIT_PATH from kernel commandline and starts landlock-test.service
- landlock-test.service starts bash uml-init.sh
- uml-init.sh extracts commandline from kernel's /proc/cmdline (the part after --) and runs it

This dependency makes it complicated to run tests in a container, or on OSes with a different init system.

Replace the init with: `/bin/bash /path/to/uml-init.sh <user-provided-commandline>`

This only changes the commandline interface of `uml-init.sh`, not that of `uml-run.sh`. The only change required in the userspace setup is to mount /proc then link /dev/fd to /proc/self/fd/ for bash redirections to work.

There's just a little tweak to pass the commandline: a first -- is needed to tell the uml-run.sh script the split between {kernel+environment variables} and {usermode commandline}. Now a different -- is needed to tell the kernel the split between its argv[0] (/bin/bash) and argv[1..] (usermode commandline)

With this patch, dependencies are essentially bash + iproute2 (to set the loopback interface up in network tests), which makes it possible to run tests in a container environment without systemd.

### CODE REVIEW

**Comment by mtth-bfft** [comment]
This allows me to run tests in my setup more easily and frequently. Thought there might be a way to improve and merge this (very experimental) patch instead of keeping it locally 🤷

File: guest/uml-init.sh [hunk 1]
```diff
@@ -22,26 +22,19 @@ exit_poweroff() {
 	if [[ -n "${UML_RET:-}" ]]; then
 		echo "$1" > "${UML_RET}"
 	fi
-	exec poweroff -f
+	exec /usr/sbin/poweroff -f
```
Comment (l0kod) [inline guest/uml-init.sh:25]: Why do we need this change? What if poweroff is not in `/usr/sbin`?

File: guest/uml-init.sh [hunk 2]
```diff
@@ -22,26 +22,19 @@ exit_poweroff() {
...
-UML_EXEC="$(< /proc/cmdline)"
-UML_EXEC="${UML_EXEC#* --}"
-
-if [[ -z "${UML_EXEC}" ]]; then
-	echo "ERROR: Missing command" >&2
-	exit_poweroff 1
-fi
+# Setup usually done by an actual init system (we use bash directly because
+# we just want to run a single commandline, and this avoids adding a
+# dependency on a given init system on the host filesystem)
```
Comment (l0kod) [inline guest/uml-init.sh:35]: A lot of mount points are missing, see eb32b5073fd451d7f52abcc2f9565855028861ba

They may not be required when running a kselftest, but they are useful for debugging or launching more complex binaries.

**Review by l0kod — COMMENTED** [review]
> This dependency makes it complicated to run tests in a container, or on OSes with a different init system.

The script `docker-run.sh` is dedicated to run this script in containers. What is the issue?

FYI, this script didn't use systemd at first:
* eb32b5073fd451d7f52abcc2f9565855028861ba
* a4c14fb6a04ab040a077d620b89d6a8f4300368b

The main motivation was to use a real init system to properly reap children and to make it easier to maintain/support for different distros. I was a bit skeptical at first but using this systemd setup worked well on several distros, at least the ones I tested (see `docker-run.sh`).

Additional requirements:
* We need to make sure all the `docker-run.sh`'s containers works.
* We also need to make sure this still work with the [GitHub CI](https://github.com/landlock-lsm/rust-landlock/blob/main/.github/workflows/rust.yml).

On which distro(s) did you test this change?

**Comment by mtth-bfft** [comment]
Hi Mickaël,

Yeeeep, was so focused on running the test suite more often that I forgot to look at _this_ repo's history... 🤦 
Usecase was broken and the toolchain does a great job as-is, no need to waste more time with this, sorry.

File: guest/uml-init.sh (hunk 2 above)
Comment (mtth-bfft) [reply guest/uml-init.sh:35]: Yep, I was too focused on running e.g. `check-linux.sh`.

**Pull request closed without merging**